File: c/p002.c
#include <stdio.h>

// Find the sum of even fibonacci sequences under LIMIT
int LIMIT = 4000000;

int main()
{
    int a = 1;
    int b = 1;
    int a_old;
    int sum = 0;

    while (a <= LIMIT)
    {
        a_old = a;
        a = b;
        b += a_old;
        if (a % 2 == 0)
        {
            sum += a;
        }
    }
    printf("%i\n", sum);
}

File: python/p002.py
# Compute the sum of the even fibonacci numbers less than 4,000,000

def fib(n):
    a, b = 1, 1
    total = 0
    while a <= n:
        a, b = b, a + b
        if a % 2 == 0:
            total += a
    return total

def compute():
    return fib(4_000_000)


if __name__ == '__main__':
    print(compute())
